### DISCUSSION TITLE
export variable from stream data

### DISCUSSION THREAD

**Discussion by lfoIncluit** [lf-edge/ekuiper#1399] [Q&A]
Is there a way to set an environment variable from stream data?

Maybe using a rule action, with the porpose of using it in another rules.

**Comment by ngjaying** [comment]
No, may be you can create a sink plugin to do this. But I think setting env var should be global to all rules instead of just target to a specific rule.

**Reply by lfoIncluit** [reply]
Yes, I want it to be global. But I need to set it from stream data
